ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,169 @@
+        type=_options.UUID(),
+        help="A UUID for the new disk. If not specified oVirt will "
+             "create a new UUID.")
+
+    cmd.add_argument(
+        "--disable-backup",
+        dest="enable_backup",
+        action="store_false",
+        help="Creates a disk that can not be used for incremental backup.")
+
+    cmd.add_argument(
+        "filename",
+        type=_options.file,
+        help="Path to image (e.g. /path/to/image.raw). "
+             "Supported formats: raw, qcow2, iso.")
+
+
+def _create_disk_info(filename, dst_format):
+    # Obtain the image info dictionary.
+    img_info = _api.info(filename)
+    if img_info["format"] not in ("qcow2", "raw"):
+        raise RuntimeError(f"Unsupported image format {img_info['format']}")
+    # Obtain all reimaining fields for DiskInfo.
+    format = dst_format or img_info["format"]
+    initial_size = _api.measure(filename, format)["required"]
+    content_type = _detect_disk_content_type(
+        img_info["filename"], img_info["format"])
+    name = _get_disk_name(
+        img_info["filename"], img_info["format"], content_type)
+    disk_format = _ovirt.RAW if format == "raw" else _ovirt.COW
+    return DiskInfo(name, initial_size, img_info["virtual-size"],
+                    content_type, disk_format)
+
+
+def _detect_disk_content_type(filename, image_format):
+    """
+    Detect disk content type
+
+    ISO format structure
+    ---------------------------------------------------------------------------
+    offset    type    value       comment
+    ---------------------------------------------------------------------------
+    0x0000                        system area (e.g. DOS/MBR boot sector)
+    0x8000    int8    0x01        primary volume descriptor type code
+    0x8001    strA    "CD001"     primary volume descriptor indentifier
+    0x8006    int8    0x01        primary volume desctptor version
+    0x8007            0x00        unused field
+
+    See https://wiki.osdev.org/ISO_9660#Overview_and_caveats for more info.
+
+    Returns:
+        ovirtsdk4.types.DiskContentType
+    """
+    if image_format == "raw":
+        with open(filename, "rb") as file:
+            file.seek(0x8000)
+            primary_volume_descriptor = file.read(8)
+        if primary_volume_descriptor == b"\x01CD001\x01\x00":
+            return _ovirt.ISO
+    return _ovirt.DATA
+
+
+def _get_disk_name(filename, dst_format, content_type):
+    """
+    Name the disk based on image file name, format and content type.
+    """
+    basename = os.path.splitext(os.path.basename(filename))[0]
+    ext = dst_format
+    if content_type == _ovirt.ISO:
+        ext = "iso"
+    return f"{basename}.{ext}"
+
+
+def upload_disk(args):
+    with _ui.ProgressBar() as pb:
+        disk_info = _create_disk_info(args.filename, args.format)
+        con = _ovirt.connect(args)
+        with closing(con):
+            pb.phase = "creating disk"
+            disk = _ovirt.add_disk(
```

Comment:
Doesn't this also have a "description" parameter? Could be nice to add that.

Reply:
Currently description is hardcoded in `_ovirt.add_disk` as `description='Uploaded disk',`.

I don't know, I'd say that unless we add a command option to have custom descriptions (which may be nice to have), it can stay as is. In the future if we see the need, it can be easily included as an additional parameter for `add_disk`.

Reply:
We need another option to let the user add custom description for the disk, and change the default to "Uploaded by ovirt-img".

Reply:
I'll open an issue.